src/components/htmlVideoPlayer/plugin.js

```diff
@@ -348,70 +356,69 @@ define(['browser', 'require', 'events', 'apphost', 'loading', 'dom', 'playbackMa
                     hls.loadSource(url);
                     hls.attachMedia(elem);
 
-                    htmlMediaHelper.bindEventsToHlsPlayer(self, hls, elem, onError, resolve, reject);
+                    htmlMediaHelper.bindEventsToHlsPlayer(this, hls, elem, this.onError, resolve, reject);
 
-                    self._hlsPlayer = hls;
+                    this._hlsPlayer = hls;
 
                     // This is needed in setCurrentTrackElement
-                    self._currentSrc = url;
+                    this._currentSrc = url;
                 });
             });
         }
 
-        function onShakaError(event) {
-
-            var error = event.detail;
+        /**
+         * @private
+         */
+        onShakaError = (event) => {
```

Comment:
```suggestion
        onShakaError(event) {
```

Reply:
This has to bet there

Reply:
I totally now this kills the build and eslint at the time, but without it that whole PR doesn't work, that's why I said I'd be good to have class properties supported

Reply:
@cromefire Now that they are, any plans to come back to this PR and fix it up?

Reply:
Yeah probably ping me if I should forget it